applications/FSIapplication/custom_utilities/constant_relaxation_convergence_accelerator.h

```diff
@@ -0,0 +1,217 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+#if !defined(KRATOS_CONSTANT_RELAXATION_CONVERGENCE_ACCELERATOR)
+#define  KRATOS_CONSTANT_RELAXATION_CONVERGENCE_ACCELERATOR
+
+/* System includes */
+
+
+/* External includes */
+
+
+/* Project includes */
+#include "convergence_accelerator.hpp"
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Constant relaxation convergence accelerator
+ * This utility corrects the iteration guess with a constant relaxation factor
+ * @tparam TSpace Linear algebra space type
+ */
+template<class TSpace>
+class ConstantRelaxationConvergenceAccelerator: public ConvergenceAccelerator<TSpace>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION( ConstantRelaxationConvergenceAccelerator );
+
+    typedef ConvergenceAccelerator<TSpace>                                 BaseType;
+
+    typedef typename BaseType::Pointer                              BaseTypePointer;
+
+    typedef typename BaseType::VectorType                                VectorType;
+    typedef typename BaseType::VectorPointerType                  VectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Get the Default Settings object
+     * This method returns the default parameters for this convergence accelerator.
+     * Note that it is required to be static since it is called during
+     * the construction of the object so no instantation exists yet.
+     * @return Parameters Default parameters json string
+     */
+    static Parameters GetDefaultSettings()
+    {
+        Parameters default_settings(R"(
+        {
+            "solver_type": "constant_relaxation",
+            "w": 0.5
+        }
+        )");
+
+        return default_settings;
+    }
+
+    /**
+     * @brief Construct a new Constant Relaxation Convergence Accelerator object
+     * Constructor with json string settings
+     * @param rConvAcceleratorParameters json string encapsulating the settings
+     */
+    ConstantRelaxationConvergenceAccelerator(Parameters &rConvAcceleratorParameters)
+    : BaseType(),
+      mOmega([] (Parameters x) -> double {x.ValidateAndAssignDefaults(GetDefaultSettings()); return x["w"].GetDouble();} (rConvAcceleratorParameters))
```

Comment:
that looks interesting, can you please explain what this does?

Reply:
I did this to have a `const double` as relaxation factor (it is saved as a member variable) but being capable to check the defaults before initializing variable value.

This is a Lambda function that takes the input json settings as arguments, checks it against the defaults, which are stored in a static method (it needs to be static since the instantiation is not "completed" yet) an then returns the value of the relaxation factor to be the initialization value for the member variable.

Reply:
cool, what I suspected.
I think I also have a usecase for this, thats why I am asking